gpu/octree/include/pcl/gpu/octree/octree.hpp

```diff
@@ -146,13 +146,22 @@ namespace pcl
               */
             void approxNearestSearch(const Queries& queries, NeighborIndices& result) const;
 
-            /** \brief Batch exact k-nearest search on GPU for k == 1 only!
+             /** \brief Batch exact k-nearest search on GPU for k == 1 only!
               * \param[in] queries array of centers
               * \param[in] k number of neighbors (only k == 1 is supported)
               * \param[out] results array of results
               */
+            PCL_DEPRECATED(1, 14, "use nearestKSearchBatch() wihich returns square distances instead")
             void nearestKSearchBatch(const Queries& queries, int k, NeighborIndices& results) const;
 
+            /** \brief Batch exact k-nearest search on GPU for k == 1 only!
+              * \param[in] queries array of centers
+              * \param[in] k number of neighbors (only k == 1 is supported)
+              * \param[out] results array of results
+              * \param[out] sqr_distances square distances to results (must be resized to (k * num of queries) a priori!)
```

Comment:
Same comment just to make sure you see it. Is there any strong reason to force users to pre allocate this vector in advance? Why can't you do it inside the method?

Reply:
You should still edit this docstring and remove the warning 
> (must be resized to (k * num of queries) a priori!)